### PR TITLE
Reset heartbeat expiry timer if heartbeat from control plane is received

### LINKED ISSUE #417: Heartbeat timer is not refreshed if a heartbeat from SMF/SPGW-C is received
**Describe the bug**

Heartbeat timer is not refreshed if a heartbeat from SMF/SPGW-C is received. 

**To Reproduce**

I was able to reproduce it with ONF's internal mock-smf. The high-level steps to reproduce:

1. Run SMF/SPGW-C with 5s heartbeat interval.
2. Run pfcpiface with the following configuration:
```
  "enable_hbTimer": true,
  "heart_beat_interval": "6s",
  "max_req_retries": 1,
```
3. Observe that heartbeat timer is not reset. 

**Expected behavior**

The PFCP Agent should reset heartbeat timer (call `heartBeatTimer.Reset()`) if  a heartbeat from SMF/SPGW-C is received. 

**Logs**

```
INFO[2022-01-20T13:16:03Z]/pfcpiface/pfcpiface/conn.go:77 main.(*PFCPConn).startHeartBeatMonitor() Starting Heartbeat timer                     
TRAC[2022-01-20T13:16:03Z]/pfcpiface/pfcpiface/messages.go:153 main.(*PFCPConn).SendPFCPMsg() Sent Association Setup Response to 172.28.0.6:8805 
TRAC[2022-01-20T13:16:06Z]/pfcpiface/pfcpiface/messages.go:118 main.(*PFCPConn).HandlePFCPMsg() Successfully processed Heartbeat Request from 172.28.0.6:8805 nodeID: 172.28.0.6 
TRAC[2022-01-20T13:16:06Z]/pfcpiface/pfcpiface/messages.go:153 main.(*PFCPConn).SendPFCPMsg() Sent Heartbeat Response to 172.28.0.6:8805   
TRAC[2022-01-20T13:16:09Z]/pfcpiface/pfcpiface/conn.go:88 main.(*PFCPConn).startHeartBeatMonitor() HeartBeat Interval Timer Expired 172.28.0.6:8805 
TRAC[2022-01-20T13:16:09Z]/pfcpiface/pfcpiface/messages.go:153 main.(*PFCPConn).SendPFCPMsg() Sent Heartbeat Request to 172.28.0.6:8805    
TRAC[2022-01-20T13:16:11Z]/pfcpiface/pfcpiface/messages.go:118 main.(*PFCPConn).HandlePFCPMsg() Successfully processed Heartbeat Request from 172.28.0.6:8805 nodeID: 172.28.0.6 
TRAC[2022-01-20T13:16:11Z]/pfcpiface/pfcpiface/messages.go:153 main.(*PFCPConn).SendPFCPMsg() Sent Heartbeat Response to 172.28.0.6:8805   
TRAC[2022-01-20T13:16:11Z]/pfcpiface/pfcpiface/messages.go:164 main.(*PFCPConn).sendPFCPRequestMessage() Request Timeout, retriesLeft: 1              
TRAC[2022-01-20T13:16:11Z]/pfcpiface/pfcpiface/messages.go:153 main.(*PFCPConn).SendPFCPMsg() Sent Heartbeat Request to 172.28.0.6:8805    
TRAC[2022-01-20T13:16:13Z]/pfcpiface/pfcpiface/messages.go:164 main.(*PFCPConn).sendPFCPRequestMessage() Request Timeout, retriesLeft: 0              
INFO[2022-01-20T13:16:13Z]/pfcpiface/pfcpiface/conn.go:232 main.(*PFCPConn).Shutdown() Shutdown complete for 172.28.0.6:8805        
INFO[2022-01-20T13:16:13Z]/pfcpiface/pfcpiface/conn.go:84 main.(*PFCPConn).startHeartBeatMonitor() Cancel HeartBeat Timer 172.28.0.6:8805   
```

### PR DESCRIPTION
**Pull request by osinstom**
Closes #417.

### CODE REVIEW

File: pfcpiface/conn.go [hunk 1]
```diff
@@ -74,26 +77,29 @@ func (pConn *PFCPConn) startHeartBeatMonitor() {
...
+	})
+	hbLog.Infoln("Starting Heartbeat timer")
 
-	heartBeatTimer := time.NewTimer(pConn.upf.hbInterval)
+	heartBeatExpiryTimer := time.NewTicker(pConn.upf.hbInterval)
 
 	for {
 		select {
 		case <-hbCtx.Done():
 			log.Infoln("Cancel HeartBeat Timer", pConn.RemoteAddr().String())
```
Comment (pudelkoM) [inline pfcpiface/conn.go:90]: Should all `log.` calls replaced with `hbLog`?
Can we just shadow `log` above?

File: pfcpiface/conn.go [hunk 2]
```diff
@@ -74,26 +77,29 @@ func (pConn *PFCPConn) startHeartBeatMonitor() {
...
 
-			reply, timeout := pConn.sendPFCPRequestMessage(r)
-
-			if reply != nil {
-				heartBeatTimer = time.NewTimer(pConn.upf.hbInterval)
-			} else if timeout {
+			if _, timeout := pConn.sendPFCPRequestMessage(r); timeout {
+				heartBeatExpiryTimer.Stop()
 				pConn.Shutdown()
 			}
```
Comment (pudelkoM) [inline pfcpiface/conn.go:104]: Should the ticker be reset on success? We send a hb request and got a reply, so the connection is live?

Comment (pudelkoM) [reply pfcpiface/conn.go:104]: The ticker does not have to be reset, but continues sending new signals until stopped.

File: pfcpiface/conn.go (hunk 1 above)
Comment (osinstom) [reply pfcpiface/conn.go:89]: I guess the `interval` is only required when starting timer, but I modified a bit the style to make it more clean.

Comment (pudelkoM) [reply pfcpiface/conn.go:89]: I actually have no idea what the best practice around context loggers is, but I like the idea and think we should use them more often.

**Review by pudelkoM — APPROVED** [review]

**Pull request merged**